impl/maven-cli/src/main/java/org/apache/maven/cling/logging/impl/MavenSimpleConfiguration.java

```diff
@@ -37,7 +40,16 @@ public void setRootLoggerLevel(Level level) {
                     case INFO -> "info";
                     default -> "error";
                 };
-        System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", value);
+
+        String defaultLogLevelKey = "org.slf4j.simpleLogger.defaultLogLevel";
+        String current = System.getProperty(defaultLogLevelKey);
+        if (current == null) {
+            System.setProperty(defaultLogLevelKey, value);
+        } else {
```

Comment:
what if the levels are the same? Then there shouldn't be a warning.

Reply:
When the value is the same, It doesn't cause `-Dorg slf4j.SimpleLogger.DefaultLogLevel` configuration affects the maven command line configuration failure.
@elharo

Reply:
I'm not sure I follow. Can you rephrase?

Reply:
For example, we run the command  `mvn clean install -Dorg slf4j.SimpleLogger.DefaultLogLevel -X`, it will lead to effective `-X`, and `-Dorgslf4j.SimpleLogger.DefaultLogLevel` failure (I think I said against the above)

Reply:
You're probably correct, but I do not understand what you are saying.

Reply:
The `-X` option will ultimately call `MavenSimpleConfiguration#setRootLoggerLevel` which will set the `org.slf4j.simpleLogger.defaultLogLevel` system property to the value, see https://github.com/apache/maven/blob/99777ace0226034f82299898e598a7f977d525ef/impl/maven-cli/src/main/java/org/apache/maven/cling/logging/impl/MavenSimpleConfiguration.java#L40

Reply:
```
➜  maven git:(feat/MNG-8512) ✗ ~/work/git/maven/apache-maven/target/apache-maven-4.0.0-rc-3-SNAPSHOT/bin/mvn -X -Dorg.slf4j.simpleLogger.defaultLogLevel=debug

[WARNING] System property 'org.slf4j.simpleLogger.defaultLogLevel' is already set to 'debug' - ignoring log level from -X/-e/-q options
Apache Maven 4.0.0-rc-3-SNAPSHOT (cc9dc285f04d254e786b9e727ebea1d368bca400)
...
```

I think what @elharo means is that we don't want to print a warning if we're using `-X` and the system property is already set to the correct value.

Reply:
Maybe we should compare and use the _lowest_ level ? And simply ignore if the two values are the same?

Reply:
Actually, I'm not sure we want user properties to take precedence over CLI options.
We use the same behavior for color for example:
https://github.com/apache/maven/blob/bebc3d4a2e987e59aa0666f71cf3b8688dde6f7a/impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java#L209-L212

Reply:
We still need to tell the user that the two configurations conflict, but changing to the INFO level makes more sense. What do you think?
cc @gnodet @elharo

Reply:
Yes, you're right.  That could be done in a single call as `System.setProperty` returns the previous value.  So we could check and if non null and different from the new value, log something at info level.